### PR TITLE
Feature/nm 258 GitHub db api

### PR DESCRIPTION
**Pull request by urbanchelo**

### CODE REVIEW

File: controllers/teamdetail.github.controller.js [hunk 1]
```diff
@@ -183,6 +220,22 @@ function getTeamCommentsCountForInterval(team, clientId, githubUrl, dateInterval
...
+        })
+        .then(clientUuid => {
+                return teamService.getTeamByName(clientUuid, teamName)
+                    .then(teamData => {
+                        return teamData[0].teamId;
+                    })
+            }
+        )
+        .catch(error => {
+            next(error);
```
Comment (olgazavodska) [inline controllers/teamdetail.github.controller.js:236]: next is not available in this function

File: services/utils.service.js [hunk 2]
```diff
@@ -23,34 +25,65 @@ function loadSettings() {
...
+            return clientData[0].settings;
         })
-        .catch(err => {
-            handleJsonParsingError(clientId, err);
-        });     
 }
 
-function getGithubBackendUrl(clientId) {
-    return loadSettings()
+function loadSettings(clientId, mode) {
```
Comment (olgazavodska) [inline services/utils.service.js:35]: MAJOR: This function has been changed from loadSettings() without parameters to loadSettings(clientId, mode), but it is referenced from settings.controller.js as     utilsService.loadSettings()

File: controllers/teamdetail.github.controller.js [hunk 3]
```diff
@@ -1,64 +1,68 @@
...
 const defaultStartDate = new Date(defaultEndDate).setDate(new Date(defaultEndDate).getDate() - fortnights * 14);
 
 function getTeamDetailWeekly(req, res, next) {
-    const clientId = utilsService.getClientId(req);
     const startDate = req.query.startDate ? req.query.startDate : utilsService.parseTimestampToDateISO(defaultStartDate);
     const endDate = req.query.endDate ? req.query.endDate : utilsService.parseTimestampToDateISO(defaultEndDate);
-    const team = utilsService.getTeamMandatory(req);  
-    
-    let dateIntervals = utilsService.getDateIntervalsWeekly(Date.parse(startDate), Date.parse(endDate));
+    const mode = req.query.mode ? req.query.mode : defaultMode;
```
Comment (olgazavodska) [inline controllers/teamdetail.github.controller.js:14]: I would suggest to create small functionality which validates provided values (something like utilsService.getTeamMandatory(req)).

File: services/utils.service.js (hunk 2 above)
Comment (urbanchelo) [reply services/utils.service.js:35]: Done

File: controllers/teamdetail.github.controller.js (hunk 1 above)
Comment (urbanchelo) [reply controllers/teamdetail.github.controller.js]: Done

File: controllers/teamdetail.github.controller.js (hunk 3 above)
Comment (urbanchelo) [reply controllers/teamdetail.github.controller.js]: Done

File: controllers/settings.controller.js [hunk 4]
```diff
@@ -2,8 +2,9 @@ const utilsService = require('../services/utils.service');
 
 function getSettings(req, res, next) {
     const clientId = utilsService.getClientId(req);
+    const mode = utilsService.getMode(req);
 
-    utilsService.loadSettings()
+    utilsService.loadSettings(clientId, mode)
```
Comment (olgazavodska) [inline controllers/settings.controller.js:7]: There is a bug in prepareData() method. First row gets data for specific clientId, but 'setting' already contains data for specific client. It should be rewritten from:
const clientSettings = settings.clients[clientId];
to
const clientSettings = settings;

Comment (urbanchelo) [reply controllers/settings.controller.js:7]: Done

**Pull request merged**